### DISCUSSION TITLE
Relative Strength Index (RSI)

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#224] [Indicators]
Community discussion about [Relative Strength Index (RSI)](https://dotnet.stockindicators.dev/indicators/Rsi/#content)

**Comment by Jammooly1** [comment]
Hello, this library is great but is there a way to get RSI for a stock that is young (somewhat recently IPOed)? 

Even if the standard deviation of error is not optimal, it'd still be great to see some result return. Whenever the RSI result doesn't return for a young stock, I have to go on a stock platform such as E-Trade to look at its RSI over an X day period.

This also relates to some other indicators as well.

Thanks

**Reply by DaveSkender** [reply]
Can you provide the symbol of the stock so I can evaluate it?  The issue for RSI, and other indicators that use similar smoothing techniques, is that it has a warmup period of at least 100 bars before the data is accurate.  That's why we set a minimum threshold of `N+100` quotes to be provided for RSI.  See #215 for more info.

**Reply by DaveSkender** [reply]
This issue comes up occasionally in discussion (eg #590) and I can see two types of users or use cases: 1) those who care about higher accuracy, and 2) those where it’s not a problem to accept less accurate results.

I’ve designed the library towards that first use case since, in my experience, I’d rather not have my automated algo trading system use inaccurate data + I want to have the ability to handle the situation gracefully with Exceptions.

With that said, if there is enough interest from our community, I can look into adding a “[damn the torpedoes](https://www.yourdictionary.com/damn-the-torpedoes)” setting that will reduce the threshold so it essentially ignores convergence issues.

**Reply by DaveSkender** [reply]
I can’t remove the threshold entirely because there is still a minimum amount required to calculate; for example, RSI needs at least enough quotes to cover the provided `lookbackPeriods`.

**Reply by Jammooly1** [reply]
Sounds great, I'd be really interested in the damn the torpedoes setting.

An example I can provide is the new stock for Gitlab, GTLB, when I look on e-trade for an RSI in the past 14 days, it says 30.93 as of the time of this post. The library is unable to return an RSI for it.

Thanks

**Reply by DaveSkender** [reply]
I’ve added to the backlog, you can follow by subscribing to #656

**Comment by russtavellid-cell** [comment]
Could you please tell me if divergence is planned to be added? 
![photo_2025-08-24_13-04-15](https://github.com/user-attachments/assets/d37e57e4-7094-4362-b397-ae9aba3d8ced)

**Reply by DaveSkender** [reply]
We’d looked into this previously and [did not find it to be a good fit](https://github.com/DaveSkender/Stock.Indicators/discussions/906#discussioncomment-3743389) for this library, but are also open to new ideas.
